include/geometric_shapes/bodies.h

```diff
@@ -71,10 +71,29 @@ struct BoundingCylinder
   Eigen::Isometry3d pose;
   double radius;
   double length;
-
   EIGEN_MAKE_ALIGNED_OPERATOR_NEW
 };
 
+// to be able to use the more efficient Eigen::Transform::linear() instead of .rotation(), we need to check the user has
+// really passed an isometry; to avoid runtime costs, this check is only done as assert, which get compiled-out in
+// release builds
+#ifdef NDEBUG
+#define ASSERT_ISOMETRY(transform) assert(true);
+#else
+#define ASSERT_ISOMETRY(transform)                                                                                     \
+  {                                                                                                                    \
+    if ((transform).matrix()(3, 0) != 0 || (transform).matrix()(3, 1) != 0 || (transform).matrix()(3, 1) != 0 ||       \
+        (transform).matrix()(3, 3) != 1 ||                                                                             \
+        std::abs(1.0 - std::abs((transform).linear().determinant())) >= detail::ZERO)                                  \
```

Comment:
Actually you should drop the abs() here, because the determinant should be positive.
Otherwise, it's a mirroring operation!

Reply:
Just checking the determinant isn't sufficient: Let's say the matrix scales by 1.0, 2.0, 0.5 along x,y,z.
The resulting determinant is still 1.0. You actually need to check for equality of the scaling matrix to the identity.

Reply:
Okay, I rewrote the test completely, check it out now.

Reply:
And I ran test_bounding_box in debug mode locally and it worked. When I passed it a non-isometry, it failed as it should.